src/services/graphService.ts

```diff
@@ -1,11 +1,31 @@
 
+import { v4 as uuid } from 'uuid';
 import { DATABASE_NAME, Tables } from '../db/db.constants';
 import { executeQuery } from '../db/queryExecuter';
 import { Graph, GraphInDB } from '../types/graph.types';
 
 class GraphService {
     tableName = Tables.graphs;
 
+	async createGraph(graph: Graph) {
+		const query = `
+		INSERT INTO ${DATABASE_NAME}."${this.tableName}"(
+			"graphId", "dataSourceId", "graphConfig", "graphTemplate", "graphName")
+			VALUES ('${uuid()}', '${graph.dataSourceId}', '${JSON.stringify(graph.graphConfig)}', '${JSON.stringify(graph.template)}', '${graph.title}') 
+			RETURNING "graphId", "dataSourceId", "graphConfig", "graphTemplate", "graphName";
+		`;
+		const row = await executeQuery<GraphInDB>(query);
+		const createdGraph = row[0];
+		
+		return {
```

Comment:
why do you need to return all? isnt only the id enought?

Reply:
possible, I just used the returned graph for the graphsAtom as well
I can use the one that is not from the server and then simply change the id of it
tell me if u want me to do that

Reply:
I think that it is better from to get it from the client because if we change the type or something so we shouldnt be coupled to the DB